fix-679: empty lockfiles with no dependencies

verth opened the pull request:
<!-- Thanks for submitting a pull request! Please provide enough information so that others can review your pull request. The two fields below are mandatory. -->

**Summary**

No lockfile is written with empty dependencies, which causes check to fail. Act the same with manifest & lockfile, so write an empty file as-well when no dependencies are going to be installed `dependencies: {}`. 

<!-- Explain the **motivation** for making this change. What existing problem does the pull request solve? -->

This will solve: https://github.com/yarnpkg/yarn/issues/679

**Test plan**

<!-- Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots / videos if the pull request changes UI. -->

```
$ yarn init
$ yarn install
$ yarn check
$ yarn check --integrity
$ npm run test
```

bestander commented:
How about a unit test?

verth commented:
Yes, I was going to, but it was super late already, added it now. I'm sorry, should have made that clear.

bestander commented:
https://github.com/yarnpkg/yarn/pull/1712

verth commented:
Cool, cool! Will keep my eye on it

bestander commented:
Ping #1712 is merged

verth commented:
I tried to kill the commits and merge master with the test only but it was not passing. When testing install manually without dependencies it also did not write the lockfile for me (with v.0.18.0 last commit https://github.com/yarnpkg/yarn/commit/15d011274eabe89a5fe8d5d32a966ba48c92755e). 

Would you know something more by any chance, intentional? (not pushed yet to keep fix alive atm)

bestander commented:
@verth, I don't think we had intentional behavior around edge cases with no dependencies.
I think integrity check should pass for empty lockfile, so go ahead rebase and give it another try if you want

verth commented:
@bestander 

Rebased and dropped the "code fix" in PR to rely on `isSync` fix, seems like something small might still be a little off? Since rebase test it not passing, tested it manually as-well and it seemed the `yarn.lock` was not created.

bestander commented:
@verth, currently yarn does not write lockfile if there are no dependencies.
A change in install.js is needed

verth commented:
@bestander Reimplemented a fix now as-well.

My bad this was dragged out so long, thought you only wanted to keep the test in this PR, I probably misinterpreted "currently working on a fix [...] However it would be great to add your test anyway".
